### ISSUE TITLE
Bug: `no-fallthrough` doesn't account for comments

### ISSUE THREAD

**Issue by felipecrs** [eslint/eslint#19663]
### Environment

Node version: 22.15.0
npm version: 11.3.0
Local ESLint version: 9.25.1
Global ESLint version: 9.25.1
Operating System: Ubuntu 24.04.2 LTS

### What parser are you using?

Default (Espree)

### What did you do?

```js
// eslint-disable-next-line no-unused-vars
function estimateCompletion(stage) {
  switch (stage.state) {
    // BlueRunResult (implies BlueRunState == FINISHED)
    case "success":
    case "unstable":
    case "failure":
    case "unknown":
    case "aborted":
    // BlueRunState
    case "skipped": // <----- THIS LINE FAILS BUT PASSES IF I REMOVE THE COMMENT LINE ABOVE
    case "not_built": // not sure???
      return 100;
    default:
      return 0;
  }
}
```

### What did you expect to happen?

To work. This is allowed as per documentation.

### What actually happened?

It fails. If I remove the comment in the line above, it passes.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLy8gZXNsaW50LWRpc2FibGUtbmV4dC1saW5lIG5vLXVudXNlZC12YXJzXG5mdW5jdGlvbiBlc3RpbWF0ZUNvbXBsZXRpb24oc3RhZ2UpIHtcbiAgc3dpdGNoIChzdGFnZS5zdGF0ZSkge1xuICAgIC8vIEJsdWVSdW5SZXN1bHQgKGltcGxpZXMgQmx1ZVJ1blN0YXRlID09IEZJTklTSEVEKVxuICAgIGNhc2UgXCJzdWNjZXNzXCI6XG4gICAgY2FzZSBcInVuc3RhYmxlXCI6XG4gICAgY2FzZSBcImZhaWx1cmVcIjpcbiAgICBjYXNlIFwidW5rbm93blwiOlxuICAgIGNhc2UgXCJhYm9ydGVkXCI6XG4gICAgLy8gQmx1ZVJ1blN0YXRlXG4gICAgY2FzZSBcInNraXBwZWRcIjpcbiAgICBjYXNlIFwibm90X2J1aWx0XCI6IC8vIG5vdCBzdXJlPz8/XG4gICAgICByZXR1cm4gMTAwO1xuICAgIGRlZmF1bHQ6XG4gICAgICByZXR1cm4gMDtcbiAgfVxufSIsIm9wdGlvbnMiOnsicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOlsiZXJyb3IiXSwiZm9yLWRpcmVjdGlvbiI6WyJlcnJvciJdLCJnZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOlsiZXJyb3IiXSwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOlsiZXJyb3IiXSwibm8tY2xhc3MtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbXBhcmUtbmVnLXplcm8iOlsiZXJyb3IiXSwibm8tY29uZC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29uc3QtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0YW50LWJpbmFyeS1leHByZXNzaW9uIjpbImVycm9yIl0sIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6WyJlcnJvciJdLCJuby1jb250cm9sLXJlZ2V4IjpbImVycm9yIl0sIm5vLWRlYnVnZ2VyIjpbImVycm9yIl0sIm5vLWRlbGV0ZS12YXIiOlsiZXJyb3IiXSwibm8tZHVwZS1hcmdzIjpbImVycm9yIl0sIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6WyJlcnJvciJdLCJuby1kdXBlLWVsc2UtaWYiOlsiZXJyb3IiXSwibm8tZHVwZS1rZXlzIjpbImVycm9yIl0sIm5vLWR1cGxpY2F0ZS1jYXNlIjpbImVycm9yIl0sIm5vLWVtcHR5IjpbImVycm9yIl0sIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1lbXB0eS1wYXR0ZXJuIjpbImVycm9yIl0sIm5vLWVtcHR5LXN0YXRpYy1ibG9jayI6WyJlcnJvciJdLCJuby1leC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjpbImVycm9yIl0sIm5vLWZhbGx0aHJvdWdoIjpbImVycm9yIl0sIm5vLWZ1bmMtYXNzaWduIjpbImVycm9yIl0sIm5vLWdsb2JhbC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW1wb3J0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1pbnZhbGlkLXJlZ2V4cCI6WyJlcnJvciJdLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6WyJlcnJvciJdLCJuby1sb3NzLW9mLXByZWNpc2lvbiI6WyJlcnJvciJdLCJuby1taXNsZWFkaW5nLWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1uZXctbmF0aXZlLW5vbmNvbnN0cnVjdG9yIjpbImVycm9yIl0sIm5vLW5vbm9jdGFsLWRlY2ltYWwtZXNjYXBlIjpbImVycm9yIl0sIm5vLW9iai1jYWxscyI6WyJlcnJvciJdLCJuby1vY3RhbCI6WyJlcnJvciJdLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOlsiZXJyb3IiXSwibm8tcmVkZWNsYXJlIjpbImVycm9yIl0sIm5vLXJlZ2V4LXNwYWNlcyI6WyJlcnJvciJdLCJuby1zZWxmLWFzc2lnbiI6WyJlcnJvciJdLCJuby1zZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjpbImVycm9yIl0sIm5vLXNwYXJzZS1hcnJheXMiOlsiZXJyb3IiXSwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOlsiZXJyb3IiXSwibm8tdW5kZWYiOlsiZXJyb3IiXSwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOlsiZXJyb3IiXSwibm8tdW5yZWFjaGFibGUiOlsiZXJyb3IiXSwibm8tdW5zYWZlLWZpbmFsbHkiOlsiZXJyb3IiXSwibm8tdW5zYWZlLW5lZ2F0aW9uIjpbImVycm9yIl0sIm5vLXVuc2FmZS1vcHRpb25hbC1jaGFpbmluZyI6WyJlcnJvciJdLCJuby11bnVzZWQtbGFiZWxzIjpbImVycm9yIl0sIm5vLXVudXNlZC1wcml2YXRlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tdW51c2VkLXZhcnMiOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1iYWNrcmVmZXJlbmNlIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtY2F0Y2giOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1lc2NhcGUiOlsiZXJyb3IiXSwibm8td2l0aCI6WyJlcnJvciJdLCJyZXF1aXJlLXlpZWxkIjpbImVycm9yIl0sInVzZS1pc25hbiI6WyJlcnJvciJdLCJ2YWxpZC10eXBlb2YiOlsiZXJyb3IiXX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by felipecrs** [comment]
> This works as intended because, by default, this rule does not require a fallthrough comment after an empty case only if the empty case and the next case are on the same line or on consecutive lines, as stated in the description of the allowEmptyCase option:

@mdjermanovic, the thing is: the lines are consecutive, except there's a comment in between them.

Is the comment line breaking the "layout" expected?

**Comment by mdjermanovic** [comment]
`case "aborted":` is on line 9, and `case "skipped":` is on line 11, so they're not on consecutive lines as there's line 10 between them.

If you set `"allowEmptyCase": true`, they don't need to be on consecutive lines, i.e., there can be comments and/or blank lines between them.

**Comment by felipecrs** [comment]
Ok, thank you.
